mlflow/pytorch/__init__.py

```diff
@@ -67,8 +68,10 @@ def get_default_pip_requirements():
     import torchvision
 
     return [
-        "torch=={}".format(torch.__version__),
-        "torchvision=={}".format(torchvision.__version__),
+        # `torch.__version__` and `torchvision.__version__`contain a local version identifier
+        # (e.g. "cpu" in "1.9.0+cpu") which causes the installation from PyPI to fail.
+        "torch=={}".format(_strip_local_version_identifier(torch.__version__)),
+        "torchvision=={}".format(_strip_local_version_identifier(torchvision.__version__)),
```

Comment:
I've switched to using `_strip_local_version_identifier` because `import torch; torch.__version__` is more robust than `importlib_metadata.version` which may fails when a user